ff4s/src/main/scala/ff4s/History.scala

```diff
@@ -0,0 +1,89 @@
+/*
+ * Copyright 2022 buntec
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package ff4s
+
+import cats.effect.implicits._
+import cats.effect.kernel.Async
+import cats.effect.kernel.Resource
+import cats.effect.std.Dispatcher
+import cats.syntax.all._
+import fs2.concurrent.Signal
+import fs2.concurrent.SignallingRef
+import io.circe.Decoder
+import io.circe.Encoder
+import io.circe.scalajs._
+import org.scalajs.dom
+
+import scalajs.js
+
+trait History[F[_], A] {
```

Comment:
It took me forever, but I finally published this 😅 

https://github.com/armanbilge/fs2-dom/blob/ae35984e8f63b0683db0d0ff1abeeb9ad034f4f0/dom/src/main/scala/fs2/dom/History.scala

Lots of other good stuff in there :)